lambdas/drivingpermitcheck/src/main/java/uk/gov/di/ipv/cri/drivingpermit/api/service/IdentityVerificationService.java

```diff
@@ -137,27 +142,37 @@ public DocumentCheckVerificationResult verifyIdentity(
                 return result;
             }
             LOGGER.error(ERROR_DRIVING_PERMIT_CHECK_RESULT_RETURN_NULL);
-            eventProbe.counterMetric(DCS_CHECK_REQUEST_FAILED);
+            eventProbe.counterMetric(DOCUMENT_DATA_VERIFICATION_REQUEST_FAILED);
 
             result.setError(ERROR_MSG_CONTEXT);
             result.setExecutedSuccessfully(false);
-        } catch (OAuthHttpResponseExceptionWithErrorBody e) {
-            eventProbe.counterMetric(DCS_CHECK_REQUEST_FAILED);
-            // Specific exception for non-recoverable DCS related errors
+        } catch (OAuthErrorResponseException e) {
+            eventProbe.counterMetric(DOCUMENT_DATA_VERIFICATION_REQUEST_FAILED);
+            // Specific exception for all non-recoverable ThirdPartyAPI related errors
             throw e;
-        } catch (InterruptedException ie) {
-            LOGGER.error(ERROR_MSG_CONTEXT, ie);
-            Thread.currentThread().interrupt();
-            result.setError(ERROR_MSG_CONTEXT + ": " + ie.getMessage());
-            result.setExecutedSuccessfully(false);
-        } catch (Exception e) {
```

Comment:
Do we not still need the catch all exception?

Reply:
Unexpected exceptions (code errors) are now caught and logged in the handler.

All exceptions in each ThirdPartyAPIService when thrown, are caught, logged at source, then re-thrown as OAuthErrorResponseException which will be caught by the handler and correctly map the OAuth response location to common-express format.

With ThirdPartyAPIService.performDocumentCheck only throwing OAuthErrorResponseException - this enables any fall back to intercept any OAuthErrorResponseException thrown and try another API.